homeassistant/components/sensor/openexchangerates.py

```diff
@@ -0,0 +1,95 @@
+"""Support for openexchangerates.org exchange rates service."""
+from datetime import timedelta
+import logging
+import requests
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_API_KEY
+
+_RESOURCE = 'https://openexchangerates.org/api/latest.json'
+_LOGGER = logging.getLogger(__name__)
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=100)
+CONF_BASE = 'base'
+CONF_QUOTE = 'quote'
+CONF_NAME = 'name'
+DEFAULT_NAME = 'Exchange Rate Sensor'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Openexchangerates sensor."""
+    payload = config.get('payload', None)
+    rest = OpenexchangeratesData(
+        _RESOURCE,
+        config.get(CONF_API_KEY),
+        config.get(CONF_BASE, 'USD'),
+        config.get(CONF_QUOTE),
+        payload
+    )
+    rest.update()
+    add_devices([OpenexchangeratesSensor(rest,
```

Comment:
Yes, check the return value at line 29, if false, log error and return. As the http error could happen anytime, you also need to check if value is None, then return, at line 64 before doing anything else with value.